lib/rubygems/specification_policy.rb

```diff
@@ -417,6 +420,17 @@ def validate_removed_attributes # :nodoc:
     end
   end
 
+  def validate_extensions # :nodoc:
+    builder = Gem::Ext::Builder.new(@specification)
+
+    rake_extension = @specification.extensions.any? {|s| builder.builder_for(s) == Gem::Ext::RakeBuilder }
+    rake_dependency = @specification.dependencies.any? {|d| d.name == 'rake'}
+
+    warning <<-WARNING if rake_extension && !rake_dependency
+You have specified rake based extension, but rake is not added as dependency. It is recommended to add rake as a dependency since there's no guarantee rake will be already installed.
```

Comment:
Normally `rake` is put in the `Gemfile` by the default `bundle gem` generator. To prevent any confusion, maybe it'd be good to specify "added as a dependency in your gemspec file", or mention the gemspec in some other way.

Reply:
I have mentioned gemspec. Can you check again?